Index out of bounds exception thrown in GradientDescentPFunc

This is frequently encountered (about 1/3 of runs) and looks to me to be caused by a score task being submitted when there are actually no scored confs. So the straightforward fix is not to submit a score task when there are no scores to submit.

Here's the exception:

```
edu.duke.cs.osprey.parallelism.TaskExecutor$TaskException: A task failed, no new tasks can be submitted
        at edu.duke.cs.osprey.parallelism.ConcurrentTaskExecutor.recordException(ConcurrentTaskExecutor.java:106)
        at edu.duke.cs.osprey.parallelism.ConcurrentTaskExecutor.taskSuccess(ConcurrentTaskExecutor.java:60)
        at edu.duke.cs.osprey.parallelism.ThreadPoolTaskExecutor.lambda$submit$0(ThreadPoolTaskExecutor.java:105)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
        at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
        at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
        at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
        at java.base/java.util.Objects.checkIndex(Objects.java:372)
        at java.base/java.util.ArrayList.get(ArrayList.java:459)
        at edu.duke.cs.osprey.kstar.pfunc.GradientDescentPfunc.onScores(GradientDescentPfunc.java:673)
        at edu.duke.cs.osprey.kstar.pfunc.GradientDescentPfunc.lambda$compute$1(GradientDescentPfunc.java:454)
        at edu.duke.cs.osprey.parallelism.ConcurrentTaskExecutor.taskSuccess(ConcurrentTaskExecutor.java:57)
```

--- a/src/edu/duke/cs/osprey/kstar/pfunc/GradientDescentPfunc.java
+++ b/src/edu/duke/cs/osprey/kstar/pfunc/GradientDescentPfunc.java
@@ -449,10 +449,12 @@ public class GradientDescentPfunc implements PartitionFunction.WithConfDB, Parti
 						confs.add(conf.getScore());
 					}
 
-					ecalc.tasks.submit(
-						new ScoreTask(instanceIdOrThrow(), confs, new Stopwatch().start()),
-						(result) -> onScores(result.scoreWeights, result.stopwatch.getTimeS())
-					);
+					if (!confs.isEmpty()) {
+						ecalc.tasks.submit(
+								new ScoreTask(instanceIdOrThrow(), confs, new Stopwatch().start()),
+								(result) -> onScores(result.scoreWeights, result.stopwatch.getTimeS())
+						);
+					}
 
 					break;
 				}

